### ISSUE TITLE
Presto throws exception for all queries on partitioned Hive tables with "float" data type if column statistics are calculated, and partition column is not specified as a filter

### ISSUE THREAD

**Issue by baloo-bear** [prestodb/presto#10353]
Consider the following Hive table creation & insert script:

```
CREATE TABLE test (mgrs varchar(11), testVal float) PARTITIONED BY (part_digit tinyint) ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' STORED AS TEXTFILE;
INSERT INTO test PARTITION(part_digit=1) VALUES ('16REU833621',0.0);
INSERT INTO test PARTITION(part_digit=2) VALUES ('11SLT745622',14.20005);
```

We can now run the following commands in Presto CLI:
```
SHOW STATS FOR hive.default.test;
 column_name | data_size | distinct_values_count | nulls_fraction | row_count | low_value | high_value
-------------+-----------+-----------------------+----------------+-----------+-----------+------------
 mgrs        | NULL      | NULL                  | NULL           | NULL      | NULL      | NULL
 part_digit  | NULL      |                   2.0 |            0.0 | NULL      | 1         | 2
 testval     | NULL      | NULL                  | NULL           | NULL      | NULL      | NULL
 NULL        | NULL      | NULL                  | NULL           |       2.0 | NULL      | NULL

SELECT * FROM hive.default.test;
    mgrs     | testval  | part_digit
-------------+----------+------------
 11SLT745622 | 14.20005 |          2
 16REU833621 |      0.0 |          1
(2 rows)
```

But if we go back to Hive and calculate column statistics on the table:
```
ANALYZE TABLE test PARTITION(part_digit) COMPUTE STATISTICS;
ANALYZE TABLE test PARTITION(part_digit) COMPUTE STATISTICS FOR COLUMNS;
```

Then we get an error when interacting with the table in Presto UNLESS we specify the partition:
```
SHOW STATS FOR test;
Query 20180409_200805_00073_m7qye failed: Object '0' does not match type long

SELECT * FROM hive.default.test;
Query 20180409_195322_00069_m7qye failed: Object '0' does not match type long

SELECT * FROM hive.default.test WHERE part_digit=1;
    mgrs     | testval | part_digit
-------------+---------+------------
 16REU833621 |     0.0 |          1
(1 row)
```

However we can go back and change the column type to "double" in Hive:
```
ALTER TABLE test CHANGE COLUMN testval testval double;
```

...and now the Presto query works again:

```
SELECT * FROM hive.default.test;
    mgrs     | testval  | part_digit
-------------+----------+------------
 11SLT745622 | 14.20005 |          2
 16REU833621 |      0.0 |          1
(2 rows)
```

Here is the full stack trace of the "Object '0' does not match type long" exception:
```
java.lang.IllegalArgumentException: Object '0' does not match type long
	at com.facebook.presto.spi.predicate.Utils.nativeValueToBlock(Utils.java:33)
	at com.facebook.presto.hive.statistics.MetastoreHiveStatisticsProvider.lambda$getTableStatistics$9(MetastoreHiveStatisticsProvider.java:163)
	at java.util.function.BinaryOperator.lambda$minBy$0(BinaryOperator.java:59)
	at java.util.stream.ReduceOps$2ReducingSink.accept(ReduceOps.java:123)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:479)
	at java.util.stream.ReferencePipeline.min(ReferencePipeline.java:520)
	at com.facebook.presto.hive.statistics.MetastoreHiveStatisticsProvider.getTableStatistics(MetastoreHiveStatisticsProvider.java:167)
	at com.facebook.presto.hive.HiveMetadata.getTableStatistics(HiveMetadata.java:401)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorMetadata.getTableStatistics(ClassLoaderSafeConnectorMetadata.java:176)
	at com.facebook.presto.metadata.MetadataManager.getTableStatistics(MetadataManager.java:368)
	at com.facebook.presto.cost.CoefficientBasedStatsCalculator$Visitor.visitTableScan(CoefficientBasedStatsCalculator.java:154)
	at com.facebook.presto.cost.CoefficientBasedStatsCalculator$Visitor.visitTableScan(CoefficientBasedStatsCalculator.java:75)
	at com.facebook.presto.sql.planner.plan.TableScanNode.accept(TableScanNode.java:136)
	at com.facebook.presto.cost.CoefficientBasedStatsCalculator.calculateStats(CoefficientBasedStatsCalculator.java:72)
	at com.facebook.presto.cost.SelectingStatsCalculator.calculateStats(SelectingStatsCalculator.java:55)
	at com.facebook.presto.cost.CachingStatsProvider.getStats(CachingStatsProvider.java:73)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter$Visitor.isKnownPlanNodeStatsOrCost(PlanPrinter.java:1386)
	at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:498)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:449)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter$Visitor.printPlanNodesStatsAndCost(PlanPrinter.java:1376)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter$Visitor.visitTableScan(PlanPrinter.java:907)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter$Visitor.visitTableScan(PlanPrinter.java:592)
	at com.facebook.presto.sql.planner.plan.TableScanNode.accept(TableScanNode.java:136)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter.<init>(PlanPrinter.java:208)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter.<init>(PlanPrinter.java:177)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter.<init>(PlanPrinter.java:164)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter.textLogicalPlan(PlanPrinter.java:260)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter.formatFragment(PlanPrinter.java:368)
	at com.facebook.presto.sql.planner.planPrinter.PlanPrinter.textPlanFragment(PlanPrinter.java:304)
	at com.facebook.presto.sql.analyzer.QueryExplainer.getPlan(QueryExplainer.java:124)
	at com.facebook.presto.execution.PlanFlattener$FlattenedPlanFragment.fromPlanFragment(PlanFlattener.java:97)
	at com.facebook.presto.execution.PlanFlattener.lambda$flatten$0(PlanFlattener.java:67)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.facebook.presto.execution.PlanFlattener.flatten(PlanFlattener.java:68)
	at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:377)
	at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:343)
	at com.facebook.presto.execution.SqlQueryExecution.start(SqlQueryExecution.java:275)
	at com.facebook.presto.execution.QueuedExecution.lambda$start$1(QueuedExecution.java:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)`
```
Thank you

**Comment by findepi** [comment]
PR: https://github.com/prestodb/presto/pull/10431

**Comment by findepi** [comment]
Fix will be available in 0.200 presto release.
